#######################################################################
################## Get functions based on pfam domains ################
############# Read comments on the script before executing ############
#######################################################################
# Used after annotating vMAGs with ANNOTATE.sh

mkdir virtak

# Split multi-fasta genomes and proteins
# https://github.com/AleCisMar/GenomicTools/blob/main/split_multi_fasta.py

split_multi_fasta.py -i genomes/all_genomes.fna -t nt -o virtak
split_multi_fasta.py -i genomes/pharokka/phanotate.faa -t aa -o virtak

# Execute VirTaK pipeline. https://github.com/AleCisMar/VirTaK
cd virtak
ls *.fna > list.txt

conda activate virtak

VirTaK.py -l list.txt -d ~/db/virtak_db/VMR_MSL38_v1_complete_genomes -p ~/db/virtak_db/ -o virtak_results.tsv
PanPhylo.py -l list.txt -d ~/db/virtak_db/VMR_MSL38_v1_complete_genomes -o panphylo
cd panphylo
get_domain_info.py domains.txt ~/db/virtak_db/Pfam-A.seed

conda deactivate

# Among several tasks performed by VirTaK.py, pfam_scan.pl is used to get domain annotations against Pfam-A database
# Among several tasks performed by PanPhylo.py, the script reads domain annotations generated by VirTaK.py and writes the full set of unique domains present in all genomes to a file named domains.txt
# get_domain_info.py extracts accessions, descriptions and comments from Pfam-A.seed for each domain (pfam_id) in domains.txt. It writes such information to a file named domain_info.tsv

# We downloaded Pfam functional categories dataset from Jahn et al. (2019) Supplementary Data S1 (sheet: Auxiliary_genes_mapping (Fig 3)).  https://doi.org/10.1016/j.chom.2019.08.019
# The functional categories were writen to a tab-delimited file named pfam2function.tsv
# A custom script was used to map our accessions in domain_info.tsv to the accession in pfam2function.tsv. https://github.com/AleCisMar/GenomicTools/blob/main/get_pfam_functions.py

get_pfam_functions.py -i domain_info.tsv -p ~/db/pfam2function.tsv -o domain_info_with_function.tsv

# Since several domains were classified as unknown or other, or even not found in the Pfam functional categories dataset, we downloaded protein HMMs for: 
#   Anti-CRISPR-associated proteins https://github.com/boweny920/AcaFinder/tree/main/HMM/AcaHMMs
#   Antibiotic resistance genes http://dantaslab.wustl.edu/resfams/Resfams-full.hmm.gz
# Then we searched our protein sequences against both HMMs with hmmscan

conda activate hmmer

hmmpress AcaHMMs.hmm
hmmpress Resfams-full.hmm

hmmscan --tblout aca_neotenic.tblout --noali -E 1e-05 --cpu 20 AcaHMMs.hmm ../../genomes/pharokka/phanotate.faa
hmmscan --tblout resfams_neotenic.tblout --noali -E 1e-05 --cpu 20 Resfams-full.hmm ../../genomes/pharokka/phanotate.faa

conda deactivate

# The unique query_name(s) were obtained from the hmmscan output tables and mapped against the pfam_scan.pl output files (created during the VirTaK process) to obtain the name of the domains present in the proteins with significant matches in the hmm databases.
for i in $(grep -v '#' aca_neotenic.tblout | sed 's/  */\t/g' | cut -f3 | sort -u) 
    do grep $i ../*.pfamscan
done | sed 's/  */\t/g' | cut -f7 | sort -u > aca_domains_neotenic.txt

for i in $(grep -v '#' aca_neotenic.tblout | sed 's/  */\t/g' | cut -f3 | sort -u) 
    do grep $i ../*.pfamscan
done | sed 's/  */\t/g' | cut -f7 | sort -u > resfams_domains_neotenic.txt

# To search for potential auxiliary metabolic genes (AMGs) we execeuted virsorter2 on genomes/all_genomes.fna to obtain the viral-affi-contigs-for-dramv.tab file. Then we executed DRAM-v annotate and distill sccripts. 
mkdir vs2

conda activate virsorter2

virsorter run -w vs2 -i ../../genomes/all_genomes.fna --include-groups dsDNAphage,RNA,ssDNA --keep-original-seq --prep-for-dramv --provirus-off

conda deactivate

mkdir dramv
mkdir dramv/distilled

conda activate DRAM1

DRAM-v.py annotate -i vs2/for-dramv/final-viral-combined-for-dramv.fa -v vs2/for-dramv/viral-affi-contigs-for-dramv.tab -o dramv --threads 20 --verbose
DRAM-v.py distill -i dramv/annotations.tsv -o dramv/distilled

conda deactivate

# AMG category was assigned to domains present in proteins with auxiliary score 1-3 with M flag (not V, A, P, B). Flags are defined in Shaffer et al. (2020) https://doi.org/10.1093/nar/gkaa621.
# AMG, Anti-CRISPR-associated and Antibiotic resistance-associated categories where added accordingly in the domain_info_with_function.tsv file.
